src/searcher/mod.rs

```diff
@@ -349,43 +351,50 @@ impl Searcher {
         Ok(())
     }
 
+    pub fn search_slice_callback_no_path<'a, 'text, 'tree>(
+        &mut self,
+        query_context: QueryContext,
+        // slice: impl TextProvider<'a> + Parseable + 'a,
```

Comment:
After having gotten `get_captures()` working with a `RopeOrSlice` argument I took a quick stab at trying to have it take the more generic `impl TextProvider + Parseable` but ran into some (more) lifetime stuff so left it more "concrete" for now at least